code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt

```diff
@@ -178,25 +176,29 @@ internal class AnalyticsHitProcessor(
                         )
                     }
                     lastHitTimestamp = timestamp
-                    processingResult.complete(true)
+                    doneProcessingResult = true
                 }
                 in arrayOf(408, 504, 503, -1) -> {
                     Log.warning(
                         AnalyticsConstants.LOG_TAG,
                         CLASS_NAME,
                         "processHit - Retrying Analytics hit, request with url $url failed with recoverable status code ${connection.responseCode}"
                     )
-                    processingResult.complete(false)
+                    doneProcessingResult = false
                 }
                 else -> {
                     Log.warning(
                         AnalyticsConstants.LOG_TAG,
                         CLASS_NAME,
                         "processHit - Dropping Analytics hit, request with url $url failed with error and unrecoverable status code ${connection.responseCode}"
                     )
-                    processingResult.complete(true)
+                    doneProcessingResult = true
                 }
             }
+            if (doneProcessingResult) {
```

Comment:
The connection needs to be closed either way here, so can you remove this check?  If the request is retried, then a new connection is created.